core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/internal/Activator.java

```diff
@@ -69,6 +72,15 @@ public void startDeploy(BundleContext bundleContext) throws Exception {
             PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
             carbonContext.setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
             carbonContext.setTenantId(MultitenantConstants.SUPER_TENANT_ID);
+
+            Collection<ServiceReference<HashProviderFactory>> hashProviderFactoryServiceReferences =
```

Comment:
Lets extract this new segment to a private method, so its objective is clearly visible and separated.

Reply:
fixed with 7dfa83cc488faeee860d640329bbef413736d73e